### PR TITLE
Update GLOSSARY.md

### PR DESCRIPTION
**Pull request by vrbritt**
Summary of changes:
- Fixed typos
- Updated glossary **terminology** using snapctl and snapd to snaptel and snapteld, respectively
- Updated **hyperlinks** using snapctl and snapd to snaptel and snapteld, respectively

Testing done:
- Tested all hyperlinks

@intelsdi-x/snap-maintainers

### CODE REVIEW

File: docs/GLOSSARY.md [hunk 1]
```diff
@@ -71,10 +71,10 @@ Snap is simple in scope and it becomes more simple when you know the terminology
 ### Snap Telemetry 
 * The full name of the Snap project, used mostly for easy searching (like snap-telemetry.io) or hashtag (#SnapTelemetry)
 
-### `snapctl`
+### snaptel
```
Comment (kindermoumoute) [inline docs/GLOSSARY.md]: Did you intentionally remove grave accent?

File: docs/GLOSSARY.md [hunk 2]
```diff
@@ -71,10 +71,10 @@ Snap is simple in scope and it becomes more simple when you know the terminology
...
 * The command-line interface (CLI) for Snap, released as a [binary][binary]
 
-### `snapd`
+### snapteld
```
Comment (kindermoumoute) [inline docs/GLOSSARY.md]: Same question

**Comment by IRCody** [comment]
Can we make this one commit? Other than that LGTM.

**Comment by mjbrender** [comment]
@IRCody as maintainers, we can do that ourselves as of the latest GitHub release 👍 

merge-and-squash

**Comment by IRCody** [comment]
@mjbrender: Doesn't that result in not having a merge commit?

**Comment by mjbrender** [comment]
@IRCody: good call - I was wrong. If we're staying consistent with merge commits, we shouldn't use that feature 😞  cc @obourdon

**Comment by kindermoumoute** [comment]
> @mjbrender: Doesn't that result in not having a merge commit?

Is it really bad not to have this merge commit?

**Comment by mjbrender** [comment]
@kindermoumoute it's not a big deal to miss it this time, but we have stuck to using them and should remain consistent.

**Pull request merged**